sentry-ruby/lib/sentry/faraday.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module Sentry
+  module Faraday
+    OP_NAME = "http.client"
+    SPAN_ORIGIN = "auto.http.faraday"
+    BREADCRUMB_CATEGORY = "http"
+
+    module Connection
+      # Since there's no way to preconfigure Faraday connections and add our instrumentation
+      # by default, we need to extend the connection constructor and do it there
+      #
+      # @see https://lostisland.github.io/faraday/#/customization/index?id=configuration
+      def new(url = nil, options = nil)
+        super.tap do |connection|
+          # Ensure that we attach instrumentation only if the adapter is not net/http
+          # because if is is, then the net/http instrumentation will take care of it
+          if connection.builder.adapter.name != "Faraday::Adapter::NetHttp"
+            connection.request :instrumentation, name: OP_NAME, instrumenter: Instrumenter.new
+
+            # Make sure that it's going to be the first middleware so that it can capture
+            # the entire request processing involving other middlewares
+            handlers = connection.builder.handlers
+            instrumentation = handlers.pop
+
+            handlers.unshift(instrumentation)
+          end
+        end
+      end
+    end
+
+    class Instrumenter
+      attr_reader :configuration
+
+      def initialize
+        @configuration = Sentry.configuration
+      end
+
+      def instrument(op_name, env, &block)
+        return unless Sentry.initialized?
+
+        Sentry.with_child_span(op: op_name, start_timestamp: Sentry.utc_now.to_f, origin: SPAN_ORIGIN) do |sentry_span|
+          request_info = extract_request_info(env)
+
+          if propagate_trace?(request_info[:url])
+            set_propagation_headers(env[:request_headers])
```

Comment:
just to make sure, the modified `env` with the extra headers gets used in the actual request right?

Reply:
nvm there's a spec that covers this, ignore